taichi/transforms/make_adjoint.cpp

```diff
@@ -201,6 +210,11 @@ class MakeAdjoint : public IRVisitor {
       auto numerator = add(sqr(bin->lhs), sqr(bin->rhs));
       accumulate(bin->lhs, div(mul(adjoint(bin), bin->rhs), numerator));
       accumulate(bin->rhs, negate(div(mul(adjoint(bin), bin->lhs), numerator)));
+    } else if (bin->op_type == BinaryOpType::pow) {
+      // d (x ^ y) = y * x ^ (y-1)  dx  +  log(x) * x ^ y  dy
```

Comment:
`d (x ^ y) = x ^ (y-1) * (y * dx + log(x) * x * dy)` better?

Reply:
Looks correct!! Yeah, it would be great if we can reuse `x ^ (y-1)` here to save one call to `pow`. Btw, could you add a test for this as well? The file where all basic operator differentation tests live: https://github.com/taichi-dev/taichi/blob/master/tests/python/test_ad_basics.py#L139